File: 0001-two-sum/0001-two-sum.go

func twoSum(nums []int, target int) []int {
    
    dict := make(map[int]int)

    for i := 0; i < len(nums); i++ {
        dict[nums[i]] = i
    }
    
    for i := 0; i < len(nums); i++ {
        sum := target - nums[i]
        _, ok := dict[sum]
        
        if (ok && dict[sum] != i) {
            return []int{dict[sum], i}
        }
    }

    return []int{}
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numbers:dict[int, int] = {}
        val: int = 0

        for i in range(len(nums)):
            numbers[nums[i]] = i

        for i in range(len(nums)):
            val = target - nums[i]
            if numbers.get(val, 0) != 0:
                if numbers[val] != i:
                    return [numbers[val], i]
        return [0,0]